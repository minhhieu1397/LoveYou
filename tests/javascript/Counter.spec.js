import { mount } from '@vue/test-utils'
import Counter from '@/Counter.vue'
import Profile from '@/profile.vue'
import MockAdapter from 'axios-mock-adapter'
import axios from 'axios'

const mock = new MockAdapter(axios)

describe('Hello.vue', () => {

    it('increments counter', () => {
        const wrapper = mount(Counter);

        expect(wrapper.vm.counter).toBe(0);

        wrapper.find('[jest="increment-button"]').trigger('click')

        expect(wrapper.vm.counter).toBe(1);
    })

    it('check axios call add json shop', async () => {
        mock
        .onGet("api/tours")
        .reply(200, {
            tour: [
                {
                    tour_name: '1',
                    vehicle: 'a',
                    departure: '1',
                    day_night: 'a',
                    price: 'a',
                    note: 'a',
                }
            ]
        })
        console.warn( await "/api/tour")
        await axios.get("api/tours")
        .then((response) => {
            const wrapper = mount(Profile);
            wrapper.find('[jest="getAll-button"]').trigger('click');
            wrapper.vm.tour = response.data;
            expect(wrapper.vm.tour).toEqual({tour: [{tour_name: '1', vehicle: 'a', departure: '1', day_night: 'a', price: 'a', note: 'a'}]  });
        })
    })
    it('snapshhot', () => {
        const wrapper = mount(Counter);

        expect(wrapper.html()).toMatchSnapshot()

        // wrapper.find('[jest="increment-button"]').trigger('click')

        // expect(wrapper.html()).toMatchSnapshot()
        
    })
})


