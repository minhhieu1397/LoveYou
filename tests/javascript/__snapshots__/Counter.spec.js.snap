// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello.vue snapshhot 1`] = `
<div>
  <h1>Count: 0</h1> <button jest="increment-button">+1</button>
  <div><button></button></div>
</div>
`;
